### DISCUSSION TITLE
s3 sink to allow append to existing file

### DISCUSSION THREAD

**Discussion by chaporgin** [vectordotdev/vector#17212] [General]
Currently S3 sink creates a new file each time it flushes the buffer. It would be nice if it could append data to an existing file.

**Comment by bruceg** [comment]
Unfortunately, the AWS S3 API only provides for writing whole files. There is no easy way of appending to an existing object. We do have an issue surrounding letting the S3 sink be more useful at creating large objects: https://github.com/vectordotdev/vector/issues/3829
